modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -1231,12 +1231,21 @@ def reset_index(self, drop):
                     name = f"level_{i}"
                 if name in exprs:
                     raise ValueError(f"cannot insert {name}, already exists")
+                if isinstance(self.columns, MultiIndex) and not isinstance(name, tuple):
+                    name = (name, *([""] * (self.columns.nlevels - 1)))
                 exprs[name] = self.ref(c)
             for c in self.columns:
                 if c in exprs:
                     raise ValueError(f"cannot insert {c}, already exists")
                 exprs[c] = self.ref(c)
-            new_columns = Index.__new__(Index, data=exprs.keys(), dtype="O")
+            new_columns = Index.__new__(
+                Index,
+                data=exprs.keys(),
+                dtype="O",
+                name=self.columns.names
```

Comment:
depends on the `data` provided this parameter will be interpreted either as `name` (in case of 1-D indices) or as `names` (in case of MultiIndex)